provider/dynamodb.go

```diff
@@ -169,181 +218,172 @@ func (store *dynamodbOnlineStore) UpdateMetadataTable(tablename string, valueTyp
-			"Tablename": {
-				S: aws.String(tablename),
+		Key: map[string]types.AttributeValue{
+			"Tablename": &types.AttributeValueMemberS{
+				Value: tablename,
 			},
 		},
 	}
-	output_val, err := store.client.GetItem(input)
+	output_val, err := store.client.GetItem(context.TODO(), input)
 	if len(output_val.Item) == 0 {
-		return NilType, fferr.NewDatasetNotFoundError("", "", fmt.Errorf("table %s not found", tablename))
+		return nil, fferr.NewDatasetNotFoundError("", "", fmt.Errorf("table %s not found", tablename))
 	}
 	if err != nil {
 		wrappedErr := fferr.NewExecutionError(pt.DynamoDBOnline.String(), err)
 		wrappedErr.AddDetail("tablename", tablename)
-		return NilType, wrappedErr
+		return nil, wrappedErr
 	}
-	metadata_item := Metadata{}
-	err = dynamodbattribute.UnmarshalMap(output_val.Item, &metadata_item)
-	if err != nil {
+	var entry dynamodbMetadataEntry
+	if err = attributevalue.UnmarshalMap(output_val.Item, &entry); err != nil {
 		wrappedErr := fferr.NewExecutionError(pt.DynamoDBOnline.String(), err)
 		wrappedErr.AddDetail("tablename", tablename)
-		return NilType, wrappedErr
+		return nil, wrappedErr
 	}
-	return ScalarType(metadata_item.Valuetype), nil
+	tableMeta, err := entry.ToTableMetadata()
+	if err != nil {
+		return nil, fferr.NewInternalError(err)
+	}
+	return tableMeta, nil
 }
 
-func GetTablename(prefix, feature, variant string) string {
+func formatDynamoTableName(prefix, feature, variant string) string {
 	tablename := fmt.Sprintf("%s__%s__%s", sn.Custom(prefix, "[^a-zA-Z0-9_]"), sn.Custom(feature, "[^a-zA-Z0-9_]"), sn.Custom(variant, "[^a-zA-Z0-9_]"))
 	return sn.Custom(tablename, "[^a-zA-Z0-9_.\\-]")
 }
 
 func (store *dynamodbOnlineStore) GetTable(feature, variant string) (OnlineStoreTable, error) {
 	key := dynamodbTableKey{store.prefix, feature, variant}
-	typeOfValue, err := store.GetFromMetadataTable(GetTablename(store.prefix, feature, variant))
+	meta, err := store.getFromMetadataTable(formatDynamoTableName(store.prefix, feature, variant))
 	if err != nil {
 		return nil, fferr.NewDatasetNotFoundError(feature, variant, err)
 	}
-	table := &dynamodbOnlineTable{client: store.client, key: key, valueType: typeOfValue}
+	table := &dynamodbOnlineTable{client: store.client, key: key, valueType: meta.Valuetype, version: meta.Version}
 	return table, nil
 }
 
 func (store *dynamodbOnlineStore) CreateTable(feature, variant string, valueType ValueType) (OnlineStoreTable, error) {
 	key := dynamodbTableKey{store.prefix, feature, variant}
-	_, err := store.GetFromMetadataTable(GetTablename(store.prefix, feature, variant))
+	tableName := formatDynamoTableName(store.prefix, feature, variant)
+	_, err := store.getFromMetadataTable(tableName)
 	if err == nil {
 		wrapped := fferr.NewDatasetAlreadyExistsError(feature, variant, nil)
-		wrapped.AddDetail("tablename", GetTablename(store.prefix, feature, variant))
+		wrapped.AddDetail("tablename", formatDynamoTableName(store.prefix, feature, variant))
```

Comment:
nit: you have the formatted tableName just use that?

Reply:
+